cpu/kinetis_common/timer.c

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ * Copyright (C) 2014 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_kinetis_common
+ * @{
+ *
+ * @file
+ * @brief       Low-level timer driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ *
+ * @}
+ */
+
+#include <stdlib.h>
+
+#include "cpu.h"
+#include "board.h"
+#include "sched.h"
+#include "thread.h"
+#include "periph_conf.h"
+#include "periph/timer.h"
+
+#if TIMER_0_EN
+
+/** Type for timer state */
+typedef struct {
+    void (*cb)(int);
+    uint32_t value;
+} timer_conf_t;
+
+/** Timer state memory */
+static timer_conf_t config[TIMER_NUMOF];
+
+
+int timer_init(tim_t dev, unsigned int ticks_per_us, void (*callback)(int))
+{
+    PIT_Type *timer;
+
+    switch (dev) {
+#if TIMER_0_EN
+
+        case TIMER_0:
+            /* enable timer peripheral clock */
+            TIMER_0_CLKEN();
+            /* set timer's IRQ priority */
+            NVIC_SetPriority(TIMER_0_IRQ_CHAN, TIMER_IRQ_PRIO);
+            /* select timer, use channel 0 as prescaler */
+            timer = TIMER_0_DEV;
+            timer->MCR = PIT_MCR_FRZ_MASK;
```

Comment:
This allows the timers to be stopped when the device enters the Debug mode. (MKW2xD RM).

ups, PIT code is a little neglected. I wanted to remove it, because it does not fit in function to riot. Maybe I will find time to rewrite it on the weekend.